--- a/os-daq-modules/daq_afpacket.c	2019-02-04 19:53:14.405531833 -0800
+++ b/os-daq-modules/daq_afpacket.c	2019-02-04 19:59:15.589273049 -0800
@@ -708,10 +708,14 @@
                 afpc->fanout_cfg.fanout_type = PACKET_FANOUT_LB;
             else if (!strcmp(entry->value, "cpu"))
                 afpc->fanout_cfg.fanout_type = PACKET_FANOUT_CPU;
+#ifdef PACKET_FANOUT_ROLLOVER
             else if (!strcmp(entry->value, "rollover"))
                 afpc->fanout_cfg.fanout_type = PACKET_FANOUT_ROLLOVER;
+#endif
+#ifdef PACKET_FANOUT_RND
             else if (!strcmp(entry->value, "rnd"))
                 afpc->fanout_cfg.fanout_type = PACKET_FANOUT_RND;
+#endif
 #ifdef PACKET_FANOUT_QM
             else if (!strcmp(entry->value, "qm"))
                 afpc->fanout_cfg.fanout_type = PACKET_FANOUT_QM;
@@ -730,8 +734,10 @@
                 snprintf(errbuf, errlen, "%s: %s requires an argument!", __func__, entry->key);
                 goto err;
             }
+#ifdef PACKET_FANOUT_FLAG_ROLLOVER
             if (!strcmp(entry->value, "rollover"))
                 afpc->fanout_cfg.fanout_flags |= PACKET_FANOUT_FLAG_ROLLOVER;
+#endif
             else if (!strcmp(entry->value, "defrag"))
                 afpc->fanout_cfg.fanout_flags |= PACKET_FANOUT_FLAG_DEFRAG;
             else
